src/ngx_http_lua_args.c

```diff
@@ -19,6 +19,38 @@ static int ngx_http_lua_ngx_req_set_uri_args(lua_State *L);
 static int ngx_http_lua_ngx_req_get_post_args(lua_State *L);
 
 
+static ngx_inline int
+ngx_http_lua_valid_args(const u_char *src, size_t size)
```

Comment:
Use of `const` looks wrong here since you actually change the value of the `src` pointer below.

Reply:
I don't like this C function name. `args` is confusing when being used alone. How about `ngx_http_lua_is_escaped_query_string()`?